codegen/type_converter.go

```diff
@@ -346,16 +346,38 @@ func (c *TypeConverter) genConverterForMap(
 		return err
 	}
 
+	valueStruct, isStruct := toFieldType.ValueSpec.(*compile.StructSpec)
+	sourceIdentifier := fromIdentifier
 	_, isStringKey := toFieldType.KeySpec.(*compile.StringSpec)
+
 	if !isStringKey {
-		return errors.Errorf(
-			"could not convert key (%s), map is not string-keyed.",
-			toField.Name,
-		)
+		realType := compile.RootTypeSpec(toFieldType.KeySpec)
+		switch realType.(type) {
+		case
+			*compile.MapSpec,
```

Comment:
The full list is here ( https://github.com/uber/zanzibar/blob/master/codegen/thrift.go#L201-L230 )

For hygiene purposes and handling all cases it's better to panic in the `default` case and to explicitely handle the `*compile.StringSpec` case.